trestle/utils/osco.py

```diff
@@ -0,0 +1,314 @@
+
+    def __init__(self, content: Dict) -> Any:
+        """Initialize given specified args."""
+        # List of extracted rules.
+        self._instances = []
+        # List of unique config_maps.
+        self._config_maps = []
+        # Dict of unique result types.
+        self._result_types = {}
+        # Dict of benchmark for this set of rules.
+        self._benchmark = {}
+        # Dict of metadata for this set of rules.
+        self._metadata = {}
+        # Dict of raw data to be processed.
+        self._content = content
+        # Perform rules extraction.
+        self._extract_rules()
+
+    @property
+    def instances(self) -> List:
+        """Get the list of rules."""
+        return self._instances
+
+    @property
+    def benchmark(self) -> Dict:
+        """Get the benchmark info."""
+        return self._benchmark
+
+    @property
+    def metadata(self) -> Dict:
+        """Get the metadata info."""
+        return self._metadata
+
+    @property
+    def analysis(self) -> Dict:
+        """Get the analysis info."""
+        logger.debug('Rules Analysis:')
+        logger.debug(f'config_maps: {self._config_maps}')
+        logger.debug(f'dispatched rules: {len(self._instances)}')
+        logger.debug(f'result types: {self._result_types}')
+        return {
+            'config_maps': self._config_maps,
+            'dispatched_rules': len(self._instances),
+            'result_types': self._result_types
+        }
+
+    def _extract_rules(self) -> None:
+        """Extract the rules from the input data."""
+        while True:
+            resource = self._content
+            if resource is None:
+                logger.debug(f'resource: {resource}')
+                break
+            if 'kind' not in resource.keys():
+                logger.debug('kind: not found')
+                break
+            if resource['kind'] != 'ConfigMap':
+                logger.debug(f'kind: {resource["kind"]}')
+                break
+            if 'results' not in resource['data']:
+                logger.debug('results: not found')
+                break
+            if 'metadata' not in resource.keys():
+                logger.debug('metadata: not found')
+                self._metadata['name'] = None
+                self._metadata['namespace'] = None
+            else:
+                self._metadata['name'] = resource['metadata']['name']
+                self._metadata['namespace'] = resource['metadata']['namespace']
+            results = resource['data']['results']
+            if results.startswith('<?xml'):
+                pass
+            else:
+                results = bz2.decompress(base64.b64decode(results))
+            logger.debug('========== <results> ==========')
+            logger.debug(results)
+            logger.debug('========== </results> ==========')
+            self._parse_xml(results, self._metadata['name'])
+            break
```

Comment:
while True is an awkward construct here for something that is non-looping. You can just
```python:
if bad_condition:
     return
``` 
instead.

Reply:
I dislike more than one exit point, but I understand the awkwardness and will comply.

Reply:
Fixed.